sympy/physics/mechanics/body.py

```diff
@@ -214,3 +214,93 @@ def apply_torque(self, vec):
         if not isinstance(vec, Vector):
             raise TypeError("A Vector must be supplied to add torque.")
         self.loads.append((self.frame, vec))
+
+    def vel(self, body=None):
+        """
+        Returns the velocity of self's masscenter wrt body's referenceframe or
+        self's referenceframe.
+
+        Parameters
+        ==========
+
+        body: Body, optional
+            The body about which the velocity is needed to be calculated.
+            If body is self by default.
+
+        Example
+        =======
+
+        >>> from sympy.physics.mechanics import Body, ReferenceFrame
+        >>> A = Body('A')
+        >>> A.masscenter.set_vel(A.frame, 5*A.frame.x)
+        >>> A.vel()
+        5*A_frame.x
+
+        >>> N = ReferenceFrame('N')
+        >>> B = Body('B', frame=N)
+        >>> A.masscenter.set_vel(N, 5*N.x)
+        >>> A.vel(B)
+        5*N.x
+
+        """
+
+        if body is None:
+            body = self
+        frame = body.frame
+        return self.masscenter.vel(frame)
+
+    def ang_vel_in(self, body):
+        """
+        Returns the angular velocity of self's frame wrt body's referenceframe or
+        self's referenceframe.
+
+        Parameters
+        ==========
+
+        body: Body
+            The body about which the angluar velocity is needed to be calculated.
+            If body is self by default.
+
+        Example
+        =======
+
+        >>> from sympy.physics.mechanics import Body, ReferenceFrame
+        >>> A = Body('A')
+        >>> N = ReferenceFrame('N')
+        >>> B = Body('B', frame=N)
+        >>> A.frame.set_ang_vel(N, 5*N.x)
+        >>> A.ang_vel_in(B)
+        5*N.x
+
+        """
+
+        return self.frame.ang_vel_in(body.frame)
+
+    def dcm(self, body):
```

Comment:
Also a body or a frame.

Reply:
I think it should be body only, because in `Joints` we would always be having 2 bodies.

Reply:
I think users (including me) will be more fluid about the use of frames and bodies. Both will be present in different scripts and there is not reason not to accept frames in all these methods. If it accepts both, then it simplifies setting up problems. A frame and a body are quite synonymous. In fact, in my course I teach that. Please make these methods accept frames or bodies.

Reply:
Done